homeassistant/components/matter/water_valve.py

```diff
@@ -0,0 +1,224 @@
+                else None
+            )
+
+            LOGGER.debug(
+                "Current position for %s - raw: %s - corrected: %s",
+                self.entity_id,
+                current_valve_position,
+                self.current_valve_position,
+            )
+
+        self._attr_device_class = ValveDeviceClass.WATER
+        supported_features = ValveEntityFeature.OPEN | ValveEntityFeature.CLOSE
+        commands = self.get_matter_attribute_value(
+            clusters.ValveConfigurationAndControl.Attributes.AcceptedCommandList
+        )
+        if clusters.ValveConfigurationAndControl.Commands.Open.command_id in commands:
+            supported_features |= ValveEntityFeature.SET_POSITION
+        self._attr_supported_features = supported_features
+
+    @callback
+    def _calculate_features(
+        self,
+    ) -> None:
+        """Calculate features for HA Valve platform from Matter FeatureMap."""
+        feature_map = int(
+            self.get_matter_attribute_value(
+                clusters.ValveConfigurationAndControl.Attributes.FeatureMap
+            )
+        )
+
+        self._feature_map = feature_map
+
+        # NOTE: the featuremap can dynamically change, so we need to update the
+        # supported features if the featuremap changes.
+        # work out supported features and presets from matter featuremap
+        if self._feature_map == feature_map:
+            return
+        self._attr_supported_features = ValveEntityFeature(0)
+        if feature_map & ValveConfigurationAndControlFeature.kLevel:
+            self._attr_supported_features |= ValveEntityFeature.SET_POSITION
+
+        # TODO: add feature for TimeSync
+        # if feature_map & ValveConfigurationAndControlFeature.TimeSync:
+        #    self._attr_supported_features |= ValveEntityFeature.TIMESYNC
+        self._attr_supported_features |= (
+            ValveEntityFeature.CLOSE | ValveEntityFeature.OPEN
+        )
+
+
+# Discovery schema(s) to map Matter Attributes to HA entities
+DISVALVEY_SCHEMAS = [
+    MatterDiscoverySchema(
+        platform=Platform.VALVE,
+        entity_description=ValveEntityDescription(
+            key="MatterValve", translation_key="valve"
+        ),
+        entity_class=MatterValve,
+        required_attributes=(
+            clusters.ValveConfigurationAndControl.Attributes.OperationalStatus,
+            clusters.ValveConfigurationAndControl.Attributes.OpenDuration,
+            clusters.ValveConfigurationAndControl.Attributes.DefaultOpenDuration,
+            clusters.ValveConfigurationAndControl.Attributes.RemainingDuration,
+            clusters.ValveConfigurationAndControl.Attributes.CurrentState,
+            clusters.ValveConfigurationAndControl.Attributes.TargetState,
+        ),
+        absent_attributes=(
+            clusters.ValveConfigurationAndControl.Attributes.CurrentLevel,
+            clusters.ValveConfigurationAndControl.Attributes.TargetLevel,
+        ),
```

Comment:
why is this ?

Reply:
I mixed it with the file used as an example: cover.py